File: combat/Action.java
package combat;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;


import joueur.Joueur;
import monstre.Monstre;


public class Action {
    
    public static ArrayList<Entite> action(ArrayList<Monstre> monstres, ArrayList<Joueur> joueurs) {
        
        ArrayList<Entite> entites =  new  ArrayList<>();

        for(Joueur joueur : joueurs)
        {
            if(!joueur.isMort()){
                entites.add(joueur);
            }
        }

        for(Monstre monstre : monstres)
        {
            if (!monstre.isMort()) {
                entites.add(monstre);
            }
        }
             
        Comparator<Entite> comparator = new Comparator<Entite>() {
            @Override
            public int compare(Entite entite1, Entite entite2) {
                float action1 = entite1.getAction();
                float action2 = entite2.getAction();
                // Compare les stats d'action
                return Float.compare(action1, action2);
            }
        };

        Collections.sort(entites, comparator);

        for(Entite entite : entites){
            System.out.println(entite.getNom() + " a " + entite.getAction() + " actions.");
        }
        
        return entites;
    }
}

File: Cs/combat/Action.cs
using System;
using System.Collections.Generic;

using joueur;
using monstre;

namespace combat
{
    public class Action
    {
        public static List<Entite> Execute(List<Monstre> monstres, List<Joueur> joueurs)
        {
            List<Entite> entites = new List<Entite>();

            foreach (Joueur joueur in joueurs)
            {
                if (!joueur.IsMort())
                {
                    entites.Add(joueur);
                }
            }

            foreach (Monstre monstre in monstres)
            {
                if (!monstre.IsMort())
                {
                    entites.Add(monstre);
                }
            }

            entites.Sort((entite1, entite2) =>
            {
                float action1 = entite1.GetAction();
                float action2 = entite2.GetAction();
                return action1.CompareTo(action2);
            });

            return entites;
        }
    }
}
